pkg/utils/managedresources/managedresources.go

```diff
@@ -314,6 +315,10 @@ func WaitUntilDeleted(ctx context.Context, client client.Client, namespace, name
 		}
 		return err
 	}
+
+	if err := kubernetesutils.WaitUntilResourcesDeleted(ctx, c, &corev1.SecretList{}, IntervalWait, client.InNamespace(namespace), client.MatchingLabels(map[string]string{v1alpha1.ReferencedBy: name})); err != nil {
```

Comment:
Nit: `client.MatchingLabels` is type alias for `map[string]string`. You can do:

```suggestion
	if err := kubernetesutils.WaitUntilResourcesDeleted(ctx, c, &corev1.SecretList{}, IntervalWait, client.InNamespace(namespace), client.MatchingLabels{v1alpha1.ReferencedBy: name}); err != nil {
```